frontend/src/routes/account/+page.server.ts

```diff
@@ -1,24 +1,21 @@
 import type { PageServerLoad } from './$types';
-import { hasPublicKey, getKeyCreationDate } from '$lib/server/database';
 import { error } from '@sveltejs/kit';
-import { getEmail } from '$lib/utils/getEmail';
 
-export const load: PageServerLoad = async ({ parent, cookies }) => {
+export const load: PageServerLoad = async ({ parent, url }) => {
 	const { session } = await parent();
 	if (!session) {
 		return { session, hasKey: false, keyCreationDate: '' };
 	}
 
-	const sessionCookie = cookies.get('user_session');
-	const user_email = await getEmail(sessionCookie ?? '');
+	const host = url.origin;
 
-	const keyResponse = await hasPublicKey(user_email);
+	const keyResponse = await fetch(`${host}/api/users/has-public-key`);
```

Comment:
Something like that would work, but idk if it is secure :D
```suggestion
	const keyResponse = await fetch(`${host}/api/users/has-public-key`, {
		headers: {
            cookie: `user_session=${cookies.get('user_session')}`
        }
	});
```

Reply:
found a solution